### ISSUE TITLE
RecursionError when calling get_yaxis_transform() on a Symlog-scaled axis.

### ISSUE THREAD

**Issue by themightyoarfish** [matplotlib/matplotlib#11163]
### Bug report

**Bug summary**

Calling `get_xaxis_transform()` after `set_yscale('symlog')` results in a `RecursionError`. I found this bug when investigating why there's no yticks (except 0) shown when using a symlog scale.

**Code for reproduction**

```python
from matplotlib import pyplot as plt
import numpy as np

f, ax = plt.subplots()
ax.plot(np.arange(100))
ax.set_yscale('symlog')
print(ax.get_yaxis_transform())
```

**Actual outcome**
```
RecursionError: Maximum recursion depth exceeded
```

**Expected outcome**
An object of type `matplotlib.scale.SymmetricLogScaleTransform` or something like that.

**Matplotlib version**

  * Operating system: Ubuntu 16.04.1
  * Matplotlib version:  2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.6.5

Installation was done via conda default channel.

**Comment by ImportanceOfBeingErnest** [comment]
Just to clarify: The code itself runs fine. The error comes from string representation of the transforms, i.e.
```
f, ax = plt.subplots()
ax.plot(np.arange(100))
ax.set_yscale('symlog')
print(ax.get_yaxis_transform())
```

will give

```
  File "...\lib\site-packages\matplotlib\transforms.py", line 1643, in __repr__
    return str(self)
RuntimeError: maximum recursion depth exceeded while getting the str of an object

```

**Comment by jklymak** [comment]
See fix in #11171, but the real fix is probably deeper...
